be/src/geo/geo_tobinary.cpp

```diff
@@ -93,6 +93,7 @@ bool toBinary::writeGeoPolygon(doris::GeoPolygon* polygon, ToBinaryContext* ctx)
     for (int i = 0; i < coordss->list.size(); ++i) {
         writeCoordinateList(*coordss->list[i], true, ctx);
     }
+    delete coordss;
```

Comment:
Not fix it in this way. Should refactor the code in GeoPolygon and return a unique ptr here.
Current code may memory leak if user call return before line 96